### PR TITLE
hann_window extension

### PR DESCRIPTION
**Pull request by nassimberrada**

### CODE REVIEW

File: ivy/functional/backends/torch/statistical.py [hunk 1]
```diff
@@ -293,3 +293,24 @@ def einsum(
     out: Optional[torch.Tensor] = None,
 ) -> torch.Tensor:
     return torch.einsum(equation, *operands)
+
+
+def hann_window(
```
Comment (Ishticode) [inline ivy/functional/backends/torch/statistical.py:298]: It seems you have implemented for torch already in statistical so can safely remove this

**Review by Ishticode — CHANGES_REQUESTED** [review]
Very good PR.

Some comments were add in certain places. Additionally, as the window_length is a type hint, it maybe appropriate to generate using st.integers() or helper.ints functions and dtype can be separately generated by helper.get_dtype I think. Let me know if any of these doesn't make sense or If i am missing something. Thanks Nassim and sorry for the delay in this review. 

There should be no array method as you have correctly left out. 
For Container methods I need to check few things so leave that out for now.If test has any issues with the decorator @handle_nestable  you can remove it.

**Review by Ishticode — CHANGES_REQUESTED** [review]
So yh. We should add container instance and static method. leaving @handle_nestable on.
Could you ensure lint is fixed and the tests are passing. Mostly seems good here.

File: ivy/functional/backends/torch/extensions.py [hunk 2]
```diff
@@ -121,3 +121,24 @@ def lcm(
...
+        periodic=periodic, 
+        dtype=dtype, 
+        layout=torch.strided,
+        device=None,
+        requires_grad=None
+    )
+
+
+hann_window.support_native_out = False
+hann_window.unsupported_dtypes = ("int32", "int64")
```
Comment (Ishticode) [inline ivy/functional/backends/torch/extensions.py:144]: It seems this part about unsupported dtypes is not required. Tests pass for me. Not sure if you had something on mind here.

File: ivy/container/extensions.py [hunk 3]
```diff
@@ -333,3 +333,105 @@ def lcm(
...
+        -------
+        ret
+            The container containing the Hann windows.
+
+        Examples
+        --------
+        With one :class:`ivy.Container` input:
+
+        >>> x = ivy.Container(a=3, b=5)
+        >>> ivy.hann_window(x)
```
Comment (Ishticode) [inline ivy/container/extensions.py:427]: should be an instance method call here.   about the square brackets in the result printed below (and above), not sure where they came from... ususally container is printed with just the curly brackets

File: ivy/container/extensions.py [hunk 4]
```diff
@@ -333,3 +333,105 @@ def lcm(
...
+        -------
+        ret
+            The container that contains the Hann windows.
+
+        Examples
+        --------
+        With one :class:`ivy.Container` input:
+
+        >>> x = ivy.Container(a=3, b=5)
+        >>> ivy.hann_window(x)
```
Comment (Ishticode) [inline ivy/container/extensions.py:376]: i think this should be `ivy.Container.static_hann`

File: ivy/functional/ivy/extensions.py [hunk 5]
```diff
@@ -668,3 +668,48 @@ def lcm(
...
+        optional output array, for writing the result to.
+
+    Returns
+    -------
+    ret
+        The array containing the window.
+
+    Functional Examples
+    -------------------
+
```
Comment (Ishticode) [inline ivy/functional/ivy/extensions.py:706]: no gap here. Sorry for being picky. The numpy style docstring test gets angry otherwise.

**Review by Ishticode — CHANGES_REQUESTED** [review]
near ready to merge. Just minor few changes

**Comment by nassimberrada** [comment]
I'm gonna have to thank you for your patience, i keep making the same typos, but i'll learn eventually...hopefully 😀

**Review by Ishticode — APPROVED** [review]
Thanks @nassimberrada. Should be merged now.✌️

**Comment by nassimberrada** [comment]
close #5541

**Pull request merged**